modules/coreos-layering-configuring-on.adoc

```diff
@@ -0,0 +1,224 @@
+    renderedImagePushSecret: # <5>
+      name: builder-dockercfg-7lzwl
+  buildOutputs: # <6>
+    currentImagePullSecret:
+      name: builder-dockercfg-7lzwl
+----
+<1> Specifies the Containerfile to configure the custom layered image.
+<2> Specifies the name of the image builder to use. This must be `PodImageBuilder`.
+<3> Specifies the name of the pull secret that the MCO needs in order to pull the base operating system image from.
+<4> Specifies the image registry to push the newly-build custom layered image to. This can be any registry that your cluster has access to. This example uses the internal {product-title} registry.
+<5> Specifies the name of the push secret that the MCO needs in order to push the newly-built custom layered image to that registry.
+<6> Specifies the secret required by the image registry that the nodes need to pull the newly-built custom layered image from your registry. This should be a different secret than the one used to push the image to your repository.
+
+.. Create the `MachineOSConfig` object:
++
+[source,terminal]
+----
+$ oc create -f <file_name>.yaml
+----
++
+The build process creates a `MachineOSBuilds` object and two new pods.
++
+[source,terminal]
+----
+NAME                                                              READY   STATUS    RESTARTS   AGE
+build-rendered-layered-ad5a3cad36303c363cf458ab0524e7c0           2/2     Running   0          2m40s # <1>
+# ...
+machine-os-builder-6fb66cfb99-zcpvq                               1/1     Running   0          2m42s # <2>
+----
+<1> This is the build pod where the custom layered image is building.
+<2> This pod can be used for troubleshooting.
+
+. When the `MachineOSBuild` object has been created and is in the `READY` state, modify the node spec for the worker nodes where you want to use the new custom layered image:
++
+.. Check that the `MachineOSBuild` object is `READY`. When the `SUCCEEDED` value is `True`, the build is complete.
++
+[source,terminal]
+----
+oc get machineosbuild
+----
++
+.Example output showing that the `MachineOSBuild` object is ready
++
+[source,terminal]
+----
+NAME                                                                PREPARED   BUILDING   SUCCEEDED   INTERRUPTED   FAILED
+layered-rendered-layered-ad5a3cad36303c363cf458ab0524e7c0-builder   False      False      True        False         False
+----
+
+.. Edit the nodes where you want to deploy the custom layered image by adding a label for the machine config pool you specified in the `MachineOSConfig` object:
++
+[source,terminal]
+----
+$ oc label node <node-name> 'node-role.kubernetes.io/<mcp_name>='
+----
++
+--
+where:
++
+node-role.kubernetes.io/<mcp_name>=:: Specifies a node selector that identifies the nodes to deploy the custom layered image. 
+--
++
+When you save the changes, the MCO drains, cordons, and reboots the nodes. After the reboot, the node will be using the new custom layered image.
```

Comment:
This part is confusing.

In the example we have created a MachineOSConfig resource for the "worker" pool. Hence, the new osImage will be directly applied to all workers in this pool. No need to modify them.

We need to modify the nodes when they belong to a different pool than the one used in the MachineOSConfig, but it is not needed any modification in the node to apply the new osImage if they are already present in this pool when the MachineOSConfig resource is created.

Reply:
@sergiordlr I made this more generic. I changed Line 46 from `spec.machineConfigPool.name: worker` to `spec.machineConfigPool.name: <mcp_name>` and added  _If necessary_ to Line 91. 
https://github.com/openshift/openshift-docs/pull/75955/commits/0e5454cd55d4b64ea07266bc190bdf064a9ddb62